sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -421,6 +422,119 @@ ng_ipv6_addr_t *ng_ipv6_netif_find_best_src_addr(kernel_pid_t pid, const ng_ipv6
     return _match_prefix(pid, dest, true);
 }
 
+/* TODO: put this somewhere more central and L2 protocol dependent */
+#define IID_LEN (8)
+
+static bool _hwaddr_to_iid(uint8_t *iid, const uint8_t *hwaddr, size_t hwaddr_len)
```

Comment:
The problem is that it is not really specified anywhere that this is the general way to do it... This function basically assumes that the address is either an EUI-64, a MAC-48, one of the short address representations of IEEE 802.15.4 (with PAN or without), or the 1-bit format we used with the cc1100 in the old stack. This is however by no means generally applicable.